sklearn/utils/validation.py

```diff
@@ -1020,13 +1020,11 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
 
 
-def _check_sample_weight(sample_weight, X, dtype=None):
+def _check_sample_weight(sample_weight, X, dtype=None, return_ones=True):
```

Comment:
I don't think we should introduce the change here considering the amount of recent PRs that use `_check_sample_weight`. #15358 

Maybe we should go with the `if ...` here and make all the changes in another PR.